utils/filesystem/zipfs_test.go

```diff
@@ -0,0 +1,264 @@
+package filesystem
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/bxcodec/faker/v3"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors/errortest"
+)
+
+const zipTestFileContent = "Test names:\r\nGeorge\r\nGeoffrey\r\nGonzo"
+
+func Test_zipFs_Exists(t *testing.T) {
+	fs, zipFile, err := NewZipFileSystemFromStandardFileSystem(filepath.Join("testdata", "testunzip.zip"), NoLimits())
+	require.NoError(t, err)
+	defer func() {
+		if zipFile != nil {
+			_ = zipFile.Close()
+		}
+	}()
+	require.NotNil(t, zipFile)
+	assert.False(t, fs.Exists(faker.DomainName()))
+	// FIXME: enable when issue in afero is fixed https://github.com/spf13/afero/issues/395
+	// assert.True(t, fs.Exists(string(filepath.Separator)))
+	// assert.True(t, fs.Exists("/"))
+	assert.True(t, fs.Exists("testunzip/test.txt"))
+	assert.True(t, fs.Exists("testunzip/child.zip"))
+	assert.True(t, fs.Exists("testunzip/ไป ไหน มา.txt"))
+	require.NoError(t, zipFile.Close())
+}
+
+func Test_zipFs_Exists2(t *testing.T) {
+	// using afero test zip file
+	fs, zipFile, err := NewZipFileSystemFromStandardFileSystem(filepath.Join("testdata", "t.zip"), NoLimits())
+	require.NoError(t, err)
+	defer func() {
+		if zipFile != nil {
+			_ = zipFile.Close()
+		}
+	}()
+	require.NotNil(t, zipFile)
+	assert.False(t, fs.Exists(faker.DomainName()))
+	assert.True(t, fs.Exists(string(filepath.Separator)))
+	assert.True(t, fs.Exists("/"))
+
+	assert.True(t, fs.Exists("testDir1"))
+	assert.True(t, fs.Exists("testFile"))
+	assert.True(t, fs.Exists("testDir1/testFile"))
+	require.NoError(t, zipFile.Close())
+}
+
+func Test_zipFS_FileInfo(t *testing.T) {
+	fs, zipFile, err := NewZipFileSystemFromStandardFileSystem(filepath.Join("testdata", "testunzip.zip"), NoLimits())
+	require.NoError(t, err)
+	defer func() {
```

Comment:
Since this is a bit ugly and we would have to do it any time we used this new function, could we maybe have a closeZip() function that we can more easily defer

Reply:
good idea. will add it

Reply:
introduced a CloseableFilesystem for this